.github/workflows/publish_snapshots.yml

```diff
@@ -0,0 +1,35 @@
+name: Publish snapshots to maven
+
+on:
+  workflow_dispatch:
+  push:
+      branches:
+        - main
+        - 1.*
+        - 2.*
+
+jobs:
+  build-and-publish-snapshots:
+    runs-on: ubuntu-latest
+
+    permissions:
+      id-token: write
+      contents: write
+
+    steps:
+      - uses: actions/setup-java@v3
```

Comment:
Can we add `name` for the steps? `Setup JDK 17`

Reply:
Done!